src/cli/self_update.rs

```
@@ -0,0 +1,276 @@
+        (false, false) => {
+            println!(
+                "{}pixi is not installed in the default location:
+- Default pixi location: {}
+- Pixi location detected: {}
+\nIt can happen when pixi has been installed via a dedicated package manager (such as Homebrew on macOS).
+You can always use `pixi self-update --force` to force the update.",
+                console::style(console::Emoji("✘ ", "")).red(),
+                default_pixi_binary_path().to_str().unwrap(),
+                env::current_exe().unwrap().to_str().unwrap()
+            );
+            return Ok(());
+        }
+        (false, true) => {}
+        (true, _) => {}
+    }
+
+    // Retrieve the target version information from github.
+    let target_version_json = retrieve_target_version(&args.version).await;
+
+    // Get the target version
+    let target_version = target_version_json.tag_name.trim_start_matches('v');
+
+    // Get the current version of the pixi binary
+    let current_version = env!("CARGO_PKG_VERSION");
+    // let current_version = "0.1.0"; // NOTE(hadim): for dev purposes
+
+    // Stop here if the target version is the same as the current version
+    if target_version == current_version {
+        println!(
+            "{}pixi is already up-to-date (version {})",
+            console::style(console::Emoji("✔ ", "")).green(),
+            current_version
+        );
+        return Ok(());
+    }
+
+    println!(
+        "{}Pixi will be updated from {} to {}",
+        console::style(console::Emoji("✔ ", "")).green(),
+        current_version,
+        target_version
+    );
+
+    // Get the name of the binary to download and install based on the current platform
+    let archive_name = default_archive_name().unwrap_or_else(|| {
+        println!(
+            "{}Not default archive name for the current platform '{}'.",
+            console::style(console::Emoji("✘ ", "")).red(),
+            std::env::consts::OS
+        );
+        std::process::exit(1);
```

I am using `std::process::exit(1);` in a couple of places in the code.

Is that ok with you? I am still struggling to find the best error handling logic. I tried a bit with `miette` but could not make it work.

Also should the exit code be `0`? or something else?

---

Yeah I think it would be better to just return a miette Error instead.

---

I used `Err(miette::miette!(error_mesage))?` - it's indeed much better!